pagopa/io-app
# chore: [IOBP-503] Add payment attempts tracking

## Short description
This PR implements the logic to track payment attempts by an user.

## List of changes proposed in this pull request
- Added `attempt` to `PaymentHistory`
- Added attempt store actions into the payment flow.

## How to test
With the `io-dev-api-server`, start a new payment flow within the wallet playground. Check with reactotron that the store is correctly updated. 
Then, restart the app and check that the data is persisted between app launches.
